pkg/common/utils_test.go

```diff
@@ -542,6 +542,237 @@ func TestConvertLabelsStringToMap(t *testing.T) {
 
 }
 
+func TestConvertTagsStringToMap(t *testing.T) {
+	t.Run("parsing tags string into slice", func(t *testing.T) {
+		testCases := []struct {
+			name           string
+			tags           string
+			expectedOutput map[string]string
+			expectedError  bool
+		}{
+			{
+				name:           "should return empty slice when tags string is empty",
+				tags:           "",
+				expectedOutput: nil,
+				expectedError:  false,
+			},
+			{
+				name:           "single tag string",
+				tags:           "parent/key/value",
+				expectedOutput: map[string]string{"parent/key": "value"},
+				expectedError:  false,
+			},
+			{
+				name:           "multiple tag string",
+				tags:           "parent1/key1/value1,parent2/key2/value2",
+				expectedOutput: map[string]string{"parent1/key1": "value1", "parent2/key2": "value2"},
+				expectedError:  false,
+			},
+			{
+				name:           "multiple tags string with whitespaces gets trimmed",
+				tags:           "parent1/key1/value1, parent2/key2/value2",
+				expectedOutput: map[string]string{"parent1/key1": "value1", "parent2/key2": "value2"},
+				expectedError:  false,
+			},
+			{
+				name:           "malformed tags string (no parent_ids, keys and values)",
+				tags:           ",,",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "malformed tags string (incorrect format)",
+				tags:           "foo,bar",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "malformed tags string (missing parent_id)",
+				tags:           "parent1/key1/value1,/key2/value2",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "malformed tags string (missing key)",
+				tags:           "parent1//value1,parent2/key2/value2",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "malformed tags string (missing value)",
+				tags:           "parent1/key1/value1,parent2/key2/",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "same tag parent_id, key and value string used more than once",
+				tags:           "parent1/key1/value1,parent1/key1/value1",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+			{
+				name:           "same tag parent_id & key string used more than once",
+				tags:           "parent1/key1/value1,parent1/key1/value2",
+				expectedOutput: nil,
+				expectedError:  true,
+			},
+		}
+
+		for _, tc := range testCases {
+			t.Logf("test case: %s", tc.name)
+			output, err := ConvertTagsStringToMap(tc.tags)
+			if tc.expectedError && err == nil {
+				t.Errorf("Expected error but got none")
+			}
+			if err != nil {
+				if !tc.expectedError {
+					t.Errorf("Did not expect error but got: %v", err)
+				}
+				continue
```

Comment:
Consider just making these three distinct if statements (like the below test), rather than adding a conditional continue

```
if err == nil && tc.expectedError { ... }

if err != nil && !tc.expectedError { ... }

if err == nil && !reflect.DeepEqual(output, tc.expectedOutput) { ... }
```

Reply:
@pwschuurman I have updated the code accordingly.